content/references/markdown/index.md

```diff
@@ -4,12 +4,176 @@ topic: markdown
 next: /references/markdown/extensions/
 ---
 
+
+
 # Basic Markdown Syntax
 
 Codewars supports [Markdown][wiki-markdown].
 
 More specifically, [GitHub Flavored Markdown][gfm] (strict superset of [CommonMark][common-mark]) and few [Codewars extensions][extensions].
 
+You may find below a summary of the most used ones.
+
+
+
+
+## Emphasis
+
+```markdown
+_This is italic_
+*This is also italic*
+
+__This is bold__
+**This is bold too**
+
+___here, italic and bold___
+***here, italic and bold***
+```
+Note that `*You **can** combine them*`: *You **can** combine them*.
+
+
+
+
+## Headers
+
+<pre><code>
+# This is &lt;h1&gt;
+## This is &lt;h2&gt;
+### This is &lt;h3&gt;
+</code></pre>
+
+
+
+## Displaying code
+
+Use code spans and code blocks to show the text as is. Use code block to show the syntax highlighted code.
+
+### Code span
+
+Use single backtick (<code>`</code>) to show the text as is inline.
+
+```markdown
+Code span example: `print("example")`
+```
+
+### Code block
+
+For multiline code blocks, use triple backticks.
+
+Constraints:
+   - Put an empty line before and after the code block itself, so that it's rendered correctly
+   - The triple backticks must be alone one their line (see [extensions](./references/markdown/extensions/) for exceptions)
```

Comment:
```suggestion
```

Reply:
sure you wanna remove those? Because they are pars of the tiny tricks to write descriptions :o

Reply:
> Because they are part of the tiny tricks to write descriptions :o

I'm not sure what you mean. Code blocks should work without empty lines.

I'm not sure what you mean on the second one. Can you explain? Maybe I'm forgetting something so examples of what works and what doesn't will help.

Reply:
wth x/ I lost my answer... x// Here it goes again:

> tricks

just things that one encounters from time to time when writing comments or descriptions. Unexpected behaviors that are a hassle to avoid, like the "non-" or "mis-rendering" of code blocks, bullet points acting ([see example here](https://www.codewars.com/kata/5efdde81543b3b00153e918a/discuss#5f1bcdbfbe281b002a2e7490); code used for the related message at the end of the present message), and stuff like that. Often, I do't even know the real cause behind those troubles.

> Can you explain the second one?

See examples in the same link as above.

---

````markdown
___Ignore this message, it's a test___

---

# about bullet points


* I'd change the formatting here, grouping/adding several subpoints: To encrypt the characters, you'll have to build pairs of characters. To do so:
  * merge all the characters
  * talk about the x stuff there.
  * "add x at the end if needed"
  * then break into pairs
* formatting: for 1) and 2), add extra whitespace so that the indentation is correct
* I'd add at the end of the previous point the note about the consecutive Xs: "Don't worry about inputs with consecutive xx, you'll never encounter them during the tests". Could be enough like that.
* the last point is curre ntly ununderstandable to me x) Maybe.... "Now you insert space characters in the ciphered string so that they are matching there original position in the input string."? (this way, that makes clear what to do about the remaining ciphered letters at the same time)
* (you will never encounter 'x' in middle of message for the sake of simplicity) To remove, since talked about before.

* check you have a fixed test with a plain text like abc cdem, so that the mix of requirements about whitespaces and addint the x between consecutive letters is made clearer/tested for sure


---


* I'd change the formatting here, grouping/adding several subpoints: To encrypt the characters, you'll have to build pairs of characters. To do so:
  * merge all the characters
  * talk about the x stuff there.
  * "add x at the end if needed"
  * then break into pairs
* formatting: for 1) and 2), add extra whitespace so that the indentation is correct
* I'd add at the end of the previous point the note about the consecutive Xs: "Don't worry about inputs with consecutive xx, you'll never encounter them during the tests". Could be enough like that.
* the last point is curre ntly ununderstandable to me x) Maybe.... "Now you insert space characters in the ciphered string so that they are matching there original position in the input string."? (this way, that makes clear what to do about the remaining ciphered letters at the same time)
* (you will never encounter 'x' in middle of message for the sake of simplicity) To remove, since talked about before.

check you have a fixed test with a plain text like abc cdem, so that the mix of requirements about whitespaces and addint the x between consecutive letters is made clearer/tested for sure


---

# about code blocks


works:
```
def pythonFunc():
    return 0
```

```
def pythonFunc():
    return 0```    <<< here

wrong ending of the block
```

some text here

first line is taken for a language specific block and is swallowed:

```def pythonFunc():
    return 0
```

````

Reply:
For the bullet points rendered weirdly:
![image](https://user-images.githubusercontent.com/639336/88588522-8c48fd00-d00c-11ea-9f53-34e4c8eb5d38.png)

It's a bug in CW's CSS. It't tricky to style lists from Markdown because it can add `<p>` inside `<li>`. I'm going to fix it by using `@tailwincss/typography` plugin like this docs sometime soon.

---

For code block, I didn't understand what you meant, but the example helped.

I didn't understand because I never thought of writing that way, knowing the Markdown spec and the example seemed clear enough. I also got confused because you wrote that extensions are exceptions (it's not).

The opening triple backticks are followed by _optional_ infostring. The infostrings are usually language ids, but it can be anything.

````markdown
```def python_func():
  return 0
```
````
So the above is a code block containing just `  return 0` with `infostring` `def python_func():`.

````markdown
```
def python_func():
    return 0```
````
is invalid because the closing backticks must be on its own line.

I think we should have an example with infostring and also mention that it should be valid language id in order to get it syntax highlighted.

Reply:
I fixed the CSS bug for Markdown and it's currently on preview. Your lists example is not broken anymore: https://preview.codewars.com/kata/5efdde81543b3b00153e918a/discuss#5f1bcdbfbe281b002a2e7490

I had to change more than I wanted, so I hope I didn't break anything else.

Reply:
> I think we should have an example with infostring and also mention that it should be valid language id in order to get it syntax highlighted.

:+1: I'll take care of that later.

(I actually faced that kind of situation with newcomers who couldn't format their code correctly in the comments)